### PR TITLE
`Development`: Remove LTI 1.0 services

### PR DESCRIPTION
**Pull request by basak-akan**
> [!WARNING]  
> **DO NOT DEPLOY CONTAINS DATABASE CHANGES!!!** 

#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server

- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage). 
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security). 
- [x] I documented the Java code using JavaDoc style.
#### Client

- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
 
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German. 

### Motivation and Context

LTI 1.0 and 1.1 is deprecated and[ 1EdTech ended all support for LTI versions 1.0, 1.1, 1.1.1, 1.1.2, 1.2, and 2.0 on June 30, 2022](https://www.imsglobal.org/lti-security-announcement-and-deprecation-schedule#:~:text=All%20support%20for%20LTI%20versions,end%20on%20June%2030%2C%202022.).  
The 'basiclti-util' dependency, integral to our LTI 1.0 implementation, is no longer maintained.

### Description

- LTI 1.0 Services are removed.
- lti_key and lti_secret columns are dropped from online course configuration table and it's entity objects are updated accordingly
- lti_outcome_url table is dropped where we store the students launch information which is accessed via LTI 1.0 
- LTI 1.0 relevant (lti_key, lti_secret, lti10launch_url) data removed from Course Management --> LTI Configuration page

### Steps for Testing

> [!WARNING]  
> **DO NOT DEPLOY CONTAINS DATABASE CHANGES!!!** 
Prerequisites:
- Do not deploy yet since it contains database changes, needs approval to be ready for testing!
- 1 Instructor
- 1 Course where online course is enabled

1. Log in to Artemis as an instructor
2. Navigate to Course Management
3. Navigate to an online course (you can use LTI Single Configuration Test)
4.  Click on "LTI Configuration" link under course details
5. Verify you don't see LTI 1.0 tab
6. Navigate to Exercises tab under LTI Configuration 
7. Verify you don't see LTI 1.0 Launch URL
8. Click on Edit
9. Verfiy you don't see LTI 1.0 related information; LTI Key and LTI Secret

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

LTI Configuration page:
image

LTI Configuration Exercises tab:
image

LTI Configuration Edit page:
image

## Summary by CodeRabbit

- **Refactor**
	- Restructured LTI integration to prioritize LTI 1.3 over LTI 1.0, updating method names and parameters accordingly.
	- Streamlined LTI configuration display in the course management UI for improved usability.
- **Bug Fixes**
	- Improved the organization and visibility of LTI configuration settings in the UI for better user experience.
- **Chores**
	- Conducted codebase cleanup by removing redundant code elements like unused imports, methods, and fields.
	- Adjusted tests to align with modifications in LTI configuration and online course configurations.

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/course/CourseTestService.java [hunk 1]
```diff
@@ -3082,7 +3076,7 @@ public void testUpdateCourseRemoveOnlineCourseConfiguration() throws Exception {
     public void testDeleteCourseDeletesOnlineConfiguration() throws Exception {
         Course course = CourseFactory.generateCourse(null, ZonedDateTime.now().minusDays(1), ZonedDateTime.now(), new HashSet<>(), "student", "tutor", "editor", "instructor");
         course.setOnlineCourse(true);
-        CourseFactory.generateOnlineCourseConfiguration(course, "test", "secret", "prefix", null);
+        CourseFactory.generateOnlineCourseConfiguration(course, "prefix", null);
```
[the remark this answers is not in the document]
Comment (basak-akan) [reply src/test/java/de/tum/in/www1/artemis/course/CourseTestService.java:3079]: generateOnlineCourseConfiguration method is intentionally changed to remove Lti 1.0 related parameters

File: src/test/java/de/tum/in/www1/artemis/course/CourseTestService.java [hunk 2]
```diff
@@ -3103,24 +3097,16 @@ public void testUpdateInvalidOnlineCourseConfiguration() throws Exception {
...
-        CourseFactory.updateOnlineCourseConfiguration(ocConfiguration, null, "secret", "prefix", null, "10000");
-        request.putWithResponseBody(getUpdateOnlineCourseConfigurationPath(courseId), ocConfiguration, OnlineCourseConfiguration.class, HttpStatus.BAD_REQUEST);
-
-        // with invalid online course configuration - no secret
-        CourseFactory.updateOnlineCourseConfiguration(ocConfiguration, "key", null, "prefix", null, "10000");
-        request.putWithResponseBody(getUpdateOnlineCourseConfigurationPath(courseId), ocConfiguration, OnlineCourseConfiguration.class, HttpStatus.BAD_REQUEST);
-
         // with invalid user prefix - not matching regex
-        CourseFactory.updateOnlineCourseConfiguration(ocConfiguration, "key", "secret", "with space", null, "10000");
+        CourseFactory.updateOnlineCourseConfiguration(ocConfiguration, "with space", null, "10000");
```
[the remark this answers is not in the document]
Comment (basak-akan) [reply src/test/java/de/tum/in/www1/artemis/course/CourseTestService.java:3102]: updateOnlineCourseConfiguration method is intentionally changed to remove Lti 1.0 related parameters

**Review by nityanandaz — APPROVED** [review]
Testing session: demonstrated locally, successfully

**Review by Kroko-fant — APPROVED** [review]
Tested in a testing session on local artemis setup (due to database changes)

**Review by Jan-Thurner — APPROVED** [review]
The Code looks good 👍

**Review by reschandreas — APPROVED** [review]
code lgtm

**Pull request merged**